sycl/doc/design/OptionalDeviceFeatures.md

```diff
@@ -913,211 +913,253 @@ aspects in the array.
+native offload bundles.  We use the same target names accepted by
+`-fsycl-targets`.  For example, if DPC++ is invoked with
+`-fsycl-targets=intel_gpu_pvc,intel_gpu_dg1`, then the driver invokes
+`clang-offload-wrapper` twice, once with `-target=intel_gpu_pvc` and once with
+`-target=intel_gpu_dg1`.
+
+As noted earlier, the user may specify a custom configuration file to compile
+for an Intel GPU device that has not yet been publicly released.  When this
+happens, the configuration file causes the driver to pass
+`clang-offload-wrapper` an option of the form `-target=intel_gpu_XXX` where the
+"XXX" represents the Intel GMDID of the device.
+
+**TODO**: We are considering moving away from the `-fsycl-targets` option and
+using the clang option `--offload-arch` instead.  If we do this, the target
+device names might change too in order to align with the naming style used by
+other offload architectures.  I presume we want to use the same strings for
+both `--offload-arch` and for the `-target` option to `clang-offload-wrapper`.
+Therefore, the target names presented above might change.
+
+**TODO**: The handling of the SPIR-V offload bundle requires some more thought.
+In the current design, there are two different types of SPIR-V modules, which
+are not exactly the same.
+
+If the user compiles with `-fsycl-targets=spir64` (which is the default), the
+device compiler is invoked with `-triple spir64-unknown-unknown`, and the
+generated IR is converted to SPIR-V.  This type of SPIR-V module is wrapped
+into the host executable via `clang-offload-wrapper -target=spir64` (i.e. the
+offload bundle has type "spir64").
+
+However, if the user AOT compiles for an Intel GPU with
+`-fsycl-targets=spir64_gen`, the device compiler is invoked with
+`-triple spir64_gen-unknown-unknown`, which is a different triple from the case
+above.  This IR is also converted to SPIR-V and then sent to the `ocloc` tool.
+The `ocloc` tool compiles the SPIR-V to native code but also embeds the SPIR-V
+into the ocloc output.  This SPIR-V is used by the device driver if the
+application is run on a GPU device that does not match any of the AOT targets.
+
+In the new design, `ocloc` does not embed SPIR-V in its output because we
+invoke it with `-exclude_ir`.  Instead, DPC++ is responsible for embedding the
+SPIR-V in the host executable, but should that SPIR-V be generated with the
+triple `spir64-unknown-unknown` or the triple `spir64_gen-unknown-unknown`?
+We could generate both, but that seems wasteful.  If we generate the SPIR-V
+with `spir64-unknown-unknown`, does that mean we need to run the device
+compiler twice, once with `spir64_gen-unknown-unknown` (to generate SPIR-V that
+is fed to `ocloc`) and once with `spir64-unknown-unknown` (to generate SPIR-V
+that is embedded in the host executable)?  This also seems wasteful because
+it requires running the device compiler twice whenever the user AOT compiles
+for an Intel GPU.
```

Comment:
I expect that we will be encouraging our users to move away from `-fsycl-targets=spir64_gen -Xs "-device pvc"` towards `-fsycl-targets=spir64_gen_pvc`. In that case, situation you describing here would more rather an exception.

We also call the device compiler twice right now for `-fsycl-targets=spir64,spir64_gen` if I'm not mistaken.

Considering those two facts, I wouldn't worry too much about this case. I think that we can leave it as-is, slowly deprecate and then completely (or almost completely) drop

Reply:
> I expect that we will be encouraging our users to move away from `-fsycl-targets=spir64_gen -Xs "-device pvc"` towards `-fsycl-targets=spir64_gen_pvc`. In that case, situation you describing here would more rather an exception.

The situation I describe occurs with the new command line syntax.  Consider a simple case where the user passes `-fsycl-targets=intel_gpu_pvc`.  This will cause the compiler to embed both a PVC and a SPIR-V image in the ELF file.  What triple will we use to generate that SPIR-V, `spir64-unknown-unknown` or `spir64_gen-unknown-unknown`?